### PR TITLE
Handle 'q' to interrumpt prompt in newshell ##newshell

### LINKED ISSUE #16668: Fix 'q' command in cfg.newshell
### Work environment

| Questions                                            | Answers
|------------------------------------------------------|--------------------
| OS/arch/bits (mandatory)                             | Debian arm 64, Ubuntu x86 32
| File format of the file you reverse (mandatory)      | PE, ELF etc.
| Architecture/bits of the file (mandatory)            | PPC, x86/32, x86/64 etc.
| r2 -v full output, **not truncated** (mandatory)     | radare2 2.4.0-git 17284 @ darwin-x86-64 git.2.2.0-476-gf8cf84e06 commit: f8cf84e0653642d9ad34e760e0e56dd81860e799 build: 2018-02-17__11:08:27

### Expected behavior
It exit.

### Actual behavior
It does not exit

### Steps to reproduce the behavior 
```
> e cfg.newshell=true
> q
```

### PR DESCRIPTION
**Pull request by ret2libc**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md#code-style)
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [radare2 book](https://github.com/radareorg/radare2book) with the relevant information (if needed)

**Detailed description**

Before this patch the code was assuming a command could return only true or false, but it is also necessary to handle a third state, which makes the prompt stop.

**Test plan**

Just try `q` or `quit` with `cfg.newshell=true`.

**Closing issues**

closes #16668

### CODE REVIEW

File: libr/core/cmd.c [hunk 1]
```diff
@@ -5044,7 +5052,14 @@ DEFINE_HANDLE_TS_FCN(arged_command) {
 }
 
 DEFINE_HANDLE_TS_FCN(legacy_quoted_command) {
-	return run_cmd_depth (state->core, node_string) != -1;
+	int res = run_cmd_depth (state->core, node_string);
```
Comment (ret2libc) [inline libr/core/cmd.c:5055]: maybe this could just be `return run_cmd_depth (state->core, node_string)` as the values are anyway the same.

Comment (ret2libc) [reply libr/core/cmd.c:5055]: actually no... :D

File: libr/core/cmd.c [hunk 2]
```diff
@@ -4602,19 +4602,27 @@ static char *ts_node_sub_parent_string(TSNode parent, TSNode node, const char *c
...
 		free (node_string);					\
 		return res;						\
 	} \
-	static bool handle_ts_##name##_internal(struct tsr2cmd_state *state, TSNode node, char *node_string)
+	static RCoreCmdStatus handle_ts_##name##_internal(struct tsr2cmd_state *state, TSNode node, char *node_string)
 
-static bool handle_ts_command(struct tsr2cmd_state *state, TSNode node);
-static bool handle_ts_command_tmpseek(struct tsr2cmd_state *state, TSNode node);
-static bool core_cmd_tsr2cmd(RCore *core, const char *cstr, bool split_lines, bool log);
+#define UPDATE_CMD_STATUS_RES(res, cmd_res, label) \
```
[the remark this answers is not in the document]
Comment (ret2libc) [reply libr/core/cmd.c:4615]: yeah it's just that if there is a `quit` command you want it to exit immediately, if a command does not work well for some reason, i try anyway to go ahead with the others. We could also stop immediately though...

**Comment by ret2libc** [comment]
> `ts_node_child_by_field_name`

Why? The field name may be useful as it could happen the index are not always the same in all types of commands.

File: libr/core/cmd.c [hunk 3]
```diff
@@ -5000,7 +5008,7 @@ DEFINE_HANDLE_TS_FCN(arged_command) {
 		char *exec_string = r_str_newf (".%s", cmd_str);
 		free (cmd_str);
 		free (command_str);
-		bool res = r_core_cmd (state->core, exec_string, false) != -1;
+		res = r_core_cmd (state->core, exec_string, false) != -1;
```
Comment (karliss) [inline libr/core/cmd.c:5011]: r_core_cmd returns the {0, -1, -2, true, false} thing. Assigning to RCoreCmdStatus might cause problems.

File: libr/core/cmd.c [hunk 4]
```diff
@@ -5034,7 +5042,7 @@ DEFINE_HANDLE_TS_FCN(arged_command) {
 	bool command_arg_space = !ts_node_is_null (args) && ts_node_end_byte (command) < ts_node_start_byte (args);
 	char *exec_string = create_exec_string (command_str, pr_args, command_arg_space);
 	R_LOG_DEBUG ("arged_command exec_string = '%s'\n", exec_string);
-	res = r_cmd_call (state->core->rcmd, exec_string) != -1;
+	res = r_cmd_call (state->core->rcmd, exec_string);
```
Comment (karliss) [inline libr/core/cmd.c:5045]: {true(1), -1}

**Comment by karliss** [comment]
> Why? The field name may be useful as it could happen the index are not always the same in all types of commands.

Aren't TSFieldId handled the same way as TSSymbol where it gets initialized together with TS language? You would still use a named variable, so the clarity of code wouldn't suffer. You might even get a compile time or at least initialization time error if the field gets renamed in syntax description file compared to child_by_field_name error that you would only get when executing specific code path.

**Comment by ret2libc** [comment]
> > Why? The field name may be useful as it could happen the index are not always the same in all types of commands.
> 
> Aren't TSFieldId handled the same way as TSSymbol where it gets initialized together with TS language? You would still use a named variable, so the clarity of code wouldn't suffer. You might even get a compile time or at least initialization time error if the field gets renamed in syntax description file compared to child_by_field_name error that you would only get when executing specific code path.

Yeah sorry I was confusing that method with `ts_node_named_child`. Yeah sounds good, thanks! In another PR.

File: libr/core/cmd.c (hunk 2 above)
[the remark this answers is not in the document]
Comment (ret2libc) [reply libr/core/cmd.c:4615]: Ok, I made it goto label even when cmd_res is INVALID. That should work as well!

**Pull request merged**